checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInferenceLogic.java

```diff
@@ -0,0 +1,284 @@
+   */
+  private void updateFieldsWithCMVsForInvocation(
+      MethodInvocationNode mNode, Set<FieldWithCMVs> fieldsWithCMVs) {
+    Node receiver = mNode.getTarget().getReceiver();
+    if (receiver.getTree() != null) {
+      Element receiverEl = TreeUtils.elementFromTree(receiver.getTree());
+      if (typeFactory.isElementPossibleOwningField(receiverEl)) {
+        Element method = TreeUtils.elementFromTree(mNode.getTree());
+        FieldWithCMVs fieldsWithCMV = getFieldsWithCMVs(receiverEl, fieldsWithCMVs);
+        if (fieldsWithCMV != null) {
+          Set<String> calledMethodsnew =
+              FluentIterable.from(fieldsWithCMV.calledMethods)
+                  .append(method.getSimpleName().toString())
+                  .toSet();
+          fieldsWithCMVs.remove(fieldsWithCMV);
+          fieldsWithCMVs.add(new FieldWithCMVs(receiverEl, calledMethodsnew));
+        } else {
+          Set<String> calledMethods = new HashSet<>();
+          calledMethods.add(method.getSimpleName().toString());
+          fieldsWithCMVs.add(new FieldWithCMVs(receiverEl, calledMethods));
+        }
+      }
+    }
+  }
+
+  /**
+   * Gets the FieldsWithCMVs whose field is equal to the elt, if one exists in {@code
+   * fieldsWithCMVs}, otherwise returns {@code null}.
+   *
+   * @param fieldsWithCMVs set of FieldWithCMVs
+   * @param elt the field element
+   * @return the FieldWithCMVs in {@code fieldsWithCMVs} whose field is equal to {@code elt}, or
+   *     {@code null} if there is no such FieldWithCMVs
+   */
+  private @Nullable FieldWithCMVs getFieldsWithCMVs(
+      Element elt, Set<FieldWithCMVs> fieldsWithCMVs) {
+    for (FieldWithCMVs fieldsWithCMV : fieldsWithCMVs) {
+      if (fieldsWithCMV.field.equals(elt)) {
+        return fieldsWithCMV;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Propagates a set of FieldWithCMVs to successors. If the successor is a regular exit point, then
+   * it searches for owning fields by looking at the called methods set stored in FieldWithCMVs and
+   * the must call obligation of each field.
+   *
+   * @param curBlock the current block
+   * @param fieldsWithCMVs the set of FieldWithCMVs for the current block
+   * @param visited block-FieldWithCMVs pairs already on the worklist
+   * @param worklist current worklist
+   */
+  private void propagateRegPaths(
+      BlockWithFieldsFacts curBlock,
+      Set<FieldWithCMVs> fieldsWithCMVs,
+      Set<BlockWithFieldsFacts> visited,
+      Deque<BlockWithFieldsFacts> worklist) {
+
+    Set<Block> successors = curBlock.block.getSuccessors();
+
+    for (Block b : successors) {
+      if (b.getType() == Block.BlockType.SPECIAL_BLOCK) {
+
+        if (((SpecialBlock) b).getSpecialType() == SpecialBlock.SpecialBlockType.EXCEPTIONAL_EXIT) {
+          // Do nothing if the successor block is an exceptional exit point
+          continue;
+        }
+
+        findOwningFields(fieldsWithCMVs, cfgMethod.getMethod());
```

Comment:
It is strange to see this call here.  From the method name, I would have thought that all this method does is propagate to successors.  Is the call here because you only want it to run when we reach the regular exit point?  If so, please add comments to that effect